sw/device/tests/flash_ctrl/flash_test.c

```diff
@@ -2,140 +2,155 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
 #include "sw/device/lib/base/stdasm.h"
 #include "sw/device/lib/common.h"
 #include "sw/device/lib/flash_ctrl.h"
-#include "sw/device/lib/gpio.h"
 #include "sw/device/lib/runtime/hart.h"
 #include "sw/device/lib/uart.h"
 
-static void break_on_error(uint32_t error) {
-  if (error) {
-    // inifinitely fetch instructions, will flag an assertion error
-    uart_send_str("FAIL!\r\n");
-    while (1) {
-      usleep(100);
-    }
+#define CHECK(cond)                                 \
+  do {                                              \
+    if (!(cond)) {                                  \
+      uart_send_str("Assertion failed at line 0x"); \
+      uart_send_uint(__LINE__, 32);                 \
+      uart_send_str("\r\nFAIL!\r\n");               \
+      abort();                                      \
+    }                                               \
+  } while (false)
+
+#define CHECK_ARRAYS_EQ(xs, ys, len) \
+  do {                               \
+    uint32_t *xs_ = (xs);            \
+    uint32_t *ys_ = (ys);            \
+    size_t len_ = (len);             \
+    for (int i = 0; i < len_; ++i) { \
+      CHECK(xs_[i] == ys_[i]);       \
+    }                                \
+  } while (false)
+
+#define CHECK_EQZ(x) CHECK((x) == 0);
+#define CHECK_NEZ(x) CHECK((x) != 0);
```

Comment:
No semicolon in single-line macro definitions.

Reply:
Oops.